pkg/apis/clusteroperator/types.go

```diff
@@ -140,6 +140,9 @@ type ClusterSpec struct {
 
 	// MachineSets specifies the configuration of all machine sets for the cluster
 	MachineSets []ClusterMachineSet
+
+	// Version references the clusterversion that should be running.
+	Version corev1.ObjectReference
```

Comment:
I don't think that we want this to be an `ObjectReference`. My understanding is that an `ObjectReference` is used when the actual object has been found. I think we want to use something akin to `LocalObjectReference`, where all we are supplying is the name and the namespace. The user should not have to supply the Kind for the object. And the user should not be confused by the presence of a UID field that does not have any effect.

Reply:
Looking at the source for the types ObjectReference was doc'd with: "ObjectReference contains enough information to let you inspect or modify the referred object", LocalObject states it's just for within one namespace and has a TODO to add Kind and UID.

Are you proposing adding our own ObjectReference type that has a namespace but no kind?

Reply:
Yes, that is precisely what I am proposing. `ObjectReference` is great in a status, where a controller is filling out the information for an existing object. In a spec, it is sub-par because it either (1) requires the user to add a lot of information that is *always* and *required* to be the same or (2) includes fields that the user *cannot* utilize and *ignored* if filled out.

Reply:
Ok that sounds fine to me. Will add.